File: Source/testPrograms/palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>
#include <stdlib.h>

bool stop(char* s){
	if(strlen(s)==3 && s[0]=='F' && s[1]=='I' && s[2]=='M') return true;
	else return false;
}

bool pali(char* s){
	int len = strlen(s)-1,
		i = 0,
		f = len;
	len = len/2 + 1;
	bool result = true;
	while(result && i!=f && i<len){
		if(s[i] != s[f]) result = false;
		i++;
		f--;
	}
	return result;
}

int main(){
	char* s;
	bool work = true;
	while(work){
		s = malloc (769 * sizeof(char));
		//fgets(s, 768, stdin);
		//scanf(" %s", s);
		scanf(" %1000[^\n]", s);
		if(stop(s)) {
			work = false;
		}
		else if(pali(s)) printf("SIM\n");
		else printf("NAO\n");
		free(s);
	}
	//free(s);
	return 0;
}

File: Source/testPrograms/palindromo.java
import java.util.Scanner;

public class palindromo{
	public static boolean stop(String s){
		if(s.length()==3 && s.charAt(0)=='F' && s.charAt(1)=='I' && s.charAt(2)=='M') return true;
		else return false;
	}

	public static boolean pali(String s){
		int i = 0,
			f = s.length()-1;
		boolean result = true;
		while(result && i!=f && i<s.length()/2){
			if(s.charAt(i) != s.charAt(f)) result = false;
			i++;
			f--;
		}
		return result;
	}

	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);
		String input = null;
		boolean stop = false;
		while(!stop){
			input = sc.nextLine();
			if(stop(input)) stop = true;
			else if(pali(input)) System.out.println("SIM");
			else System.out.println("NAO");
		}
	}
}
